### ISSUE TITLE
Improve isEmpty check of validateField function

### ISSUE THREAD

**Issue by rvetere** [react-hook-form/react-hook-form#11180]
## TLDR;
> By increasing the priority of `inputValue` in the `isEmpty` check, we can have working validation with fields that only update the state (but not the DOM). [PR](https://github.com/react-hook-form/react-hook-form/pull/11179)

### Background
First of all, this project is _absolutely_ amazing! It's simply the best form library i ever worked with! 

I've debugged the `validateField` method a lot now and I think the line "isEmpty" could be improved.

We are implementing a multi-select field with downshift.js and there we are creating custom `onChange` events that are correctly updating the internal form state of react-hook-form - everything great so far.

The problem is, we don't have a native `select` element in the DOM, representing a correct value in a classic way. 

This makes react-hook-form think the field is empty.

<img width="487" alt="Screenshot 2023-11-09 at 20 09 17" src="https://github.com/react-hook-form/react-hook-form/assets/1505976/5e0942c4-b4e0-435c-b3fc-a63bfb66bb62">

In that case, we have a value that is an array, and - as you can see, the field would have a valid value indeed - but still, react-hook-form thinks its empty.

<img width="616" alt="Screenshot 2023-11-09 at 20 09 44" src="https://github.com/react-hook-form/react-hook-form/assets/1505976/25af53fb-f3d4-4bc8-9ef6-857f1b0320d4">

This is because the "ref" is not having a value in plain HTML:

<img width="323" alt="Screenshot 2023-11-09 at 20 09 59" src="https://github.com/react-hook-form/react-hook-form/assets/1505976/b540df31-72c6-4104-92de-0808915d4ffe">

But, the last part where it checks that the value also could be an Array, is not failing:

<img width="371" alt="Screenshot 2023-11-09 at 20 10 12" src="https://github.com/react-hook-form/react-hook-form/assets/1505976/b7aafeeb-8421-4136-81d2-861c36e86dce">

This forces us to do a very ugly workaround in our implementation:

<img width="398" alt="Screenshot 2023-11-09 at 20 14 50" src="https://github.com/react-hook-form/react-hook-form/assets/1505976/8e96a9b6-cd9a-4168-8ba0-1194c1073718">

This "hack" is fixing the is-empty-check, but nobody in our company is going to understand why our MultiSelectDropdown must have this value on the "ToggleButton" ^^

It is just to suppress the ref-check.

I've created a PR that would (hopefully) improve this check: https://github.com/react-hook-form/react-hook-form/pull/11179

**Comment by bluebill1049** [comment]
Thanks very much for the PR! what's your `validateField` function looks like?

**Comment by bluebill1049** [comment]
thanks @rvetere will check it later. I think for custom-required checks, `validate` function is ideal for such, but I will verify the use case.

**Comment by bluebill1049** [comment]
parking in the todo column for now.

**Comment by rvetere** [comment]
Sorry i was in vacation the past 4 weeks - my colleagues have rewritten our "custom field implementation" in the meantime and we are not facing this issue anymore :)

**Comment by bluebill1049** [comment]
kk good to hear ad thanks for let us know. 🙏
